oaknational/Oak-Web-Application
# Feat/lesq 962/combined unit listing pages [LESQ-962]

## Description

Music year: 1994

- remove custom logic for maths on subject listing and unit listing pages and promote for all subjects
- remove legacy flag from subject listing page query and simplify
- refactor subject card double to just a single
- refactor props for subject card
- refactor unit listing page and add tests for helper methods

## How to test

note: ks4 science subjects are a bit unique, we are going to remove the singular legacy units as they are replicated in the 'new' curriculum, this means there are technically programmes which are unreachable through this new journey but that is fine (also the counts for the unit and lessons will be incorrect on the combined listing until those units are taken down)

1. Go to subject listing pages eg https://deploy-preview-2647--oak-web-application.netlify.thenational.academy/teachers/key-stages/ks2/subjects
3. You should see all subjects are combined into a single card
4. EYFS should be all legacy
5. Specialist should be unaffected  https://deploy-preview-2647--oak-web-application.netlify.thenational.academy/teachers/specialist/subjects
6. Subjects with programmes should proceed to correct programme listing page
7. Unit listing page should correctly group legacy and new units

## Screenshots

How it used to look (delete if n/a):
<img width="600" alt="Screenshot 2024-07-30 at 09 10 02" src="https://github.com/user-attachments/assets/f495a547-d8b1-44ee-a901-3de809b93231">
<img width="600" alt="Screenshot 2024-07-30 at 09 10 49" src="https://github.com/user-attachments/assets/292fd27d-d91d-460a-95aa-2f1d66348e81">


How it should now look:

<img width="600" alt="Screenshot 2024-07-30 at 09 10 30" src="https://github.com/user-attachments/assets/172ef200-931a-4d96-956e-1257c7c7ad04">
<img width="600" alt="Screenshot 2024-07-30 at 09 10 59" src="https://github.com/user-attachments/assets/b843633c-cd11-4184-90c2-dedf134249a2">

## Checklist

- [ ] Only display ‘new’ threads
- [ ] Add link for new/legacy curriculum pages
- [ ] Change subject card to be one button
